support-figma/extended-layout-plugin/src/ui.html

```diff
@@ -1114,8 +1118,39 @@
         input.addEventListener('change', function (evt) {
           const newValue = this.value;
           if (outputStringData.has(newValue)) {
-            input.value = stringRes["name"];
-            showToast(`${newValue} already exists. Revert since duplicates are not allowed.`);
+            let newStringRes = outputStringData.get(newValue);
+            let hasChanged = false;
+            // Use toString() to compare string and string array.
+            if (stringRes["text"].toString() === newStringRes["text"].toString()) {
+              let result = confirm(`Would you like to merge the translations with ${newStringRes["name"]}?`);
+              if (result) {
+                // Merge the text nodes and upate the UI.
+                for (const node of stringRes["textNodes"]) {
+                  if (newStringRes["textNodes"]) {
+                    newStringRes["textNodes"].push(node);
+                  } else {
+                    newStringRes["textNodes"] = [node];
+                  }
+                }
+                // Remove the string resource from the map.
+                outputStringData.delete(stringRes["name"]);
+                // Notify figma to update the plugin data.
+                stringRes["name"] = newValue;
+                parent.postMessage({
+                  pluginMessage: {
+                    msg: 'update-localization-data',
+                    item: stringRes
+                  }
+                }, '*');
+                // Refresh the table, which recreate the whole table...
+                createOutputStringTable('outputStrings');
+                hasChanged = true;
+              }
+            }
+            if (!hasChanged) {
+              input.value = stringRes["name"];
+              showToast(`${newValue} already exists. Revert since duplicates are not allowed.`);
```

Comment:
```suggestion
              showToast(`${newValue} already exists. Reverting resource name since duplicates are not allowed.`);
```

Reply:
Would be nice to have an option to un-merge two nodes into different string resources, but that can be done later after we get some feedback.